### PR TITLE
add mesh splitting tests

### PR DESCRIPTION
**Pull request by rwcarlsen**
follow up to #10623

### CODE REVIEW

File: test/src/userobjects/SplitTester.C [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  return validParams<GeneralUserObject>();
+  // pars.set<MultiMooseEnum>("execute_on") = "timestep_begin";
+}
+
+SplitTester::SplitTester(const InputParameters & parameters) : GeneralUserObject(parameters) {}
+
+void
+SplitTester::execute()
+{
+  auto & m = _fe_problem.mesh().getMesh();
```
Comment (permcody) [inline test/src/userobjects/SplitTester.C]: use `MooseMesh::errorIfDistributedMesh()`

File: test/src/userobjects/SplitTester.C [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+#include "MooseMesh.h"
+
+#include "libmesh/distributed_mesh.h"
+
+template <>
+InputParameters
+validParams<SplitTester>()
+{
+  return validParams<GeneralUserObject>();
+  // pars.set<MultiMooseEnum>("execute_on") = "timestep_begin";
```
Comment (permcody) [inline test/src/userobjects/SplitTester.C]: Drop

File: test/src/userobjects/SplitTester.C [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+}
+
+SplitTester::SplitTester(const InputParameters & parameters) : GeneralUserObject(parameters) {}
+
+void
+SplitTester::execute()
+{
+  auto & m = _fe_problem.mesh().getMesh();
+  if (!dynamic_cast<DistributedMesh *>(&m))
+    mooseError("not using DistributedMesh but should be");
```
Comment (permcody) [inline test/src/userobjects/SplitTester.C]: For new use Caps on error messages in MOOSE. That's the way we do it today. We can debate about switching everything to lower case, but this is inconsistent.

File: test/src/userobjects/SplitTester.C [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "SplitTester.h"
+#include "MooseMesh.h"
+
+#include "libmesh/distributed_mesh.h"
```
Comment (permcody) [inline test/src/userobjects/SplitTester.C]: You won't need this once you use the MooseMesh method.

**Review by permcody — CHANGES_REQUESTED** [review]

File: test/src/userobjects/SplitTester.C (hunk 1 above)
Comment (rwcarlsen) [reply test/src/userobjects/SplitTester.C]: We want the opposite - ``errorIfNotDistributedMesh()``.

Comment (permcody) [reply test/src/userobjects/SplitTester.C]: OK - you can still query the mesh type right from MooseMesh.

**Review by permcody — APPROVED** [review]

**Comment by permcody** [comment]
Failures are unrelated.

**Pull request merged**